client/components/domains/register-domain-step/analytics.js

```diff
@@ -95,3 +95,21 @@ export function recordShowMoreResults( searchQuery, pageNumber, section ) {
 		} )
 	);
 }
+
+export function recordFilterReset( filters, keysToReset, section ) {
+	return composeAnalytics(
+		recordGoogleEvent( 'Domain Search', 'Reset Filters' ),
+		recordTracksEvent( 'calypso_domain_search_results_filter_reset', {
+			keys_to_reset: keysToReset,
+			section,
+			...filters,
```

Comment:
This causes an error in the console log:
```
Tracks: Event `calypso_domain_search_results_filter_submit` will be rejected because property name `showExactMatchesOnly` does not match /^[a-z_][a-z0-9_]*$/. Please use a compliant property name.
```
(and same for other filters: `maxCharacters` and `includeDashes`)

Also:
```
Tracks: Unable to record event "calypso_domain_search_results_filter_submit" because nested properties are not supported by Tracks. Check 'tlds' on {...}
```

Reply:
Ah, yes. You're completely right. Luckily, this has already been fixed in #24718 (which also added analytics tracking to filter resets and submissions). In #24718, the filter keys are snake-cased to resolve this issue.

Will rebase soon.